### DISCUSSION TITLE
PushNotifications plug in problem with android

### DISCUSSION THREAD

**Discussion by ghost** [ionic-team/capacitor#3378] [Q&A]
Hello,

I am using capacitor with integration with quasar and run it as follows:
  `quasar dev -m capacitor -T android `
This means I am running it for mobile development. and I am using android virtual device/Physical device to test it.
The capacitor plugin for push notification throws a strange error **PushNotifications does not have web implementation.** 
I am not using it in web implementation so why this happens.

My capacitor config file is as follows:

```
{
  "appId": "com.frmdelivery",
  "appName": "FRM Delivery",
  "bundledWebRuntime": false,
  "npmClient": "npm",
  "webDir": "www",
  "PushNotifications": {
    "presentationOptions": [
      "badge",
      "sound",
      "alert"
    ]
  }
}
```
Would you please let me know what did I do wrong to get this result.

**Comment by imhoffd** [comment]
How are you importing the plugin?

**Comment by codenamezjames** [comment]
@pollaeng it looks like you can access capacitor from `this.$q.capacitor.Plugins.PushNotifications` Its working for me. 
but I do have an issue with 
```
PushNotifications.requestPermission().then(result => {
  if (result.granted) PushNotifications.register()
})
```
doing that crashes android. It makes it to the result.granted and crashes on register.
